src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -266,14 +269,12 @@ def _normalize(
         # If user defined their udf as @udf(lambda: x, ...) then extension_fn.handler is <lambda>.
         extension_fn.handler = f"{py_file.stem}.{extension_fn.handler}"
 
-        snowpark_lib_found = False
-        snowpark_lib_regex = re.compile(SNOWPARK_LIB_REGEX)
-        for pkg in extension_fn.packages:
-            if snowpark_lib_regex.fullmatch(ensure_string_literal(pkg.strip())):
-                snowpark_lib_found = True
-                break
-        if not snowpark_lib_found:
-            extension_fn.packages.append(SNOWPARK_LIB_NAME)
+        extension_fn.packages = [
+            self._normalize_package_name(pkg) for pkg in extension_fn.packages
+        ]
+        snowpark_lib_name = ensure_string_literal(SNOWPARK_LIB_NAME)
+        if snowpark_lib_name not in extension_fn.packages:
```

Comment:
(The packages list is not expected to be long, so this is not really an issue.) 
We're doing two linear operations here: list comprehension and the contains check. Should `_normalize_package_name` perhaps take care of both in one iteration, similar to the `snowpark_lib_found` approach on the left?

Reply:
Just found it to be cleaner like that. As you said, this is expected to be a very short list, so I'm ok with the repeated traversal.